chore: update docs for shared subscribe (#1405)
emqtt/emq-docs-cn: en_US/configuration/configuration.md

| Type | Optional Value                                                   | Default  |
| ---- | ---------------------------------------------------------------- | -------- |
| enum | `hash_clientid`, `hash_topic`, `local`, `random`, `round_robin`, `sticky`,  | - |

#### Description

Set a dispatch strategy for shared subscribers. Options are:

- **hash_clientid**: Map (hash) publisher ClientID to subscriber.
- **hash_topic**: Map (hash) source MQTT topic to subscriber.

> - **local**: Selects random subscriber connected to the node which received the message. If no such subscribers present, selects a random cluster-wise

Reviewer: ```suggestion
- **local**: Select a random subscriber in the group locally in the node which is processing the shared-publish request. If there is no such local subscriber, it fallbacks to `random` strategy 
```

- **random**: Choose randomly among all subscribers.
- **round_robin**: Enumerate subscribers in (unsorted) order.

> - **sticky**: First dispatch is random, then stick to it for all subsequent messages.

Reviewer 2: maybe `The first dispatch of each session` is better?
Or will this https://github.com/emqx/emqx/issues/9062 be a bug?
The customers think the `sticky` is the same for the same `clientid` in different sessions

Reviewer: > maybe `The first dispatch of each session` is better?
> Or will this [emqx/emqx#9062](https://github.com/emqx/emqx/issues/9062) be a bug?
> The customers think the `sticky` is the same for the same `clientid` in different sessions

Maybe add: “The stickiness lifecycle is bound to the publisher connection lifecycle, that is: if the publisher had to reconnect, stickiness will have to be reinitialized”

Author: I corrected that desc. Plz check it🤔

<br />

### broker.sample_group.shared_subscription_strategy

| Type | Optional Value                                                   | Default  |
| ---- | ---------------------------------------------------------------- | -------- |
| enum | `hash_clientid`, `hash_topic`, `local`, `random`, `round_robin`, `sticky`,  | - |

> Overrides the dispatch strategy for a shared subscription group named `sample_group`. If not configured, `broker.shared_subscription_strategy` will be used.

Reviewer: ```suggestion
Overrides the dispatch strategy for a shared subscription group named `sample_group`. If not configured, `broker.shared_subscription_strategy` is used,
```

> Where `sample_group` can be configured to any group name.

Reviewer: ```suggestion
Where `sample_group` can be configured to any group name.
```

Author: Looks the same? Did you miss something?
